pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -580,6 +580,7 @@ cdef class _Timestamp(ABCTimestamp):
 
         if own_tz is not None and not is_utc(own_tz):
             pydatetime_to_dtstruct(self, &dts)
+            # TODO: can + self.nanosecond also overflow here?
```

Comment:
Hmm OK. Do you have any idea what code paths hit this? I tried running pandas/tests/tslibs but couldn't figure out how to get here.

Happy to make sure we have proper behavior for this here or in a follow up

Reply:
I would be OK with a follow up.

I think a public way to hit this would similar to :

```python
ts = pd.Timestamp.max - pd.Timedelta(hours=8) + pd.Timedelta(some nanoseconds)
# the tz here should "add" 8 hours to get back to UTC time
# when day_name is called
ts = ts.tz_localize("US/Pacific")
ts.day_name()
```

Reply:
I _think_ that the npy_datetimestruct_to_datetime might catch that already on this PR.

```python
In [1]: import pandas as pd
In [2]: ts = pd.Timestamp(-2 ** 63 + 1).tz_localize("America/New_York")
In [3]: ts.day_name()
OverflowError                             Traceback (most recent call last)
Cell In[3], line 1
----> 1 ts.day_name()
File timestamps.pyx:805, in pandas._libs.tslibs.timestamps._Timestamp.day_name()
File timestamps.pyx:775, in pandas._libs.tslibs.timestamps._Timestamp._get_date_name_field()
File timestamps.pyx:586, in pandas._libs.tslibs.timestamps._Timestamp._maybe_convert_value_to_local()
OverflowError: Overflow occurred in npy_datetimestruct_to_datetime
````

Versus main:

```python
In [1]: import pandas as pd
In [2]: ts = pd.Timestamp(-2 ** 63 + 1).tz_localize("America/New_York")
In [3]: ts.day_name()
Out[1]: 'Tuesday'
```

Reply:
Ah okay, then I'm more confident that `+ self.nanosecond` won't overflow